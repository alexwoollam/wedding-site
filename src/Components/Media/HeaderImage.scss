.menu{
  & .row{
    max-height: 130px;
    overflow: hidden;
    & .col{
      & .media{
        & img {
          width: auto;
          max-height: 130px;
          margin: 0 auto;
        }
      }
    }
  }

}