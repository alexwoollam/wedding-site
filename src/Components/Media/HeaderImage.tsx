import React from 'react';
import './HeaderImage.scss';

interface ImageSent {
    src: string,
    alt: string
}

function HeaderImage( props:ImageSent ) {
  if( ! props.src ){
    return null;
  }

  return (
    <div className={"site-heading"}>
      <div className={"site-heading-left site-heading-side"}></div>
      <div className={"site-heading-center"}></div>
      <div className={"site-heading-right site-heading-side"}></div>
    </div>
  );
}

export default HeaderImage;