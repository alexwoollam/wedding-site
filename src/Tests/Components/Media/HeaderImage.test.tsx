import '@testing-library/jest-dom';
import React from 'react';
import { render, screen } from '@testing-library/react';
import HeaderImage from '../../../Components/Media/HeaderImage';

describe('<HeadingImage/> component', () => {
  test('Component should render', () => {
    render(<HeaderImage src="foo.png" alt="bar" />);
    const linkElement = screen.getByTestId("header-image-block");
    expect(linkElement).toBeInTheDocument();
  });

  test('Components src should be foo.png', () => {
      render(<HeaderImage src="foo.png" alt="bar" />);
      const linkElement = screen.getByRole('img', { src: 'foo.png' });
      expect(linkElement).toBeInTheDocument();
  });

  test('Components alt should be bar', () => {
      render(<HeaderImage src="foo.png" alt="bar" />);
      const linkElement = screen.getByAltText("bar");
      expect(linkElement).toBeInTheDocument();
  });
});