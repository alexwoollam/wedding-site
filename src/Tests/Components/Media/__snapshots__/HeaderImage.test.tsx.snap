// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderImage/> component renders correctly 1`] = `
<div
  className="Header flex justify-center row"
  data-testid="header-image-block"
>
  <div
    className="col-sm-12 col-md-6 m-auto col"
  >
    <div
      center="true"
      className="media"
    >
      <img
        alt="bar"
        className="media-object"
        data-src="foo.png"
      />
    </div>
  </div>
</div>
`;
